fieldservice_sale/models/sale_order.py

```diff
@@ -138,10 +138,14 @@ def _field_find_fsm_order(self):
     def _action_confirm(self):
         """On SO confirmation, some lines generate field service orders."""
         result = super(SaleOrder, self)._action_confirm()
-        if any(
-            sol.product_id.field_service_tracking != "no" for sol in self.order_line
-        ):
-            if not self.fsm_location_id:
+        fs_sale_lines = self.order_line.filtered(
+            lambda l: l.product_id.field_service_tracking != "no"
+        )
+        if fs_sale_lines:
+            if (
+                any(not sol.fsm_location_id for sol in fs_sale_lines)
+                and not self.fsm_location_id
+            ):
                 raise ValidationError(_("FSM Location must be set"))
```

Comment:
May we have a different error message when there is no FSM in one sale order line (with some hints like line number or product);